### PR TITLE
core: Rename _OpT to IRDLOperationInvT (NFC)

### PR DESCRIPTION
**Pull request by math-fehr**
This renames `_OpT` to IRDLOperationInvT, and also adds covariant and contravariant versions.

### CODE REVIEW

**Review by tobiasgrosser — APPROVED** [review]
This indeed seems more readable.

**Review by superlopuh — APPROVED** [review]

File: xdsl/irdl.py [hunk 1]
```diff
@@ -434,7 +434,11 @@ def irdl_to_attr_constraint(
 #  \___/| .__/ \___|_|  \__,_|\__|_|\___/|_| |_|
 #       |_|
 
-_OpT = TypeVar("_OpT", bound="IRDLOperation")
+IRDLOperationInvT = TypeVar("IRDLOperationInvT", bound="IRDLOperation")
+IRDLOperationCovT = TypeVar("IRDLOperationCovT", bound="IRDLOperation", covariant=True)
+IRDLOperationContrT = TypeVar(
+    "IRDLOperationContrT", bound="IRDLOperation", contravariant=True
+)
```
Comment (webmiche) [inline xdsl/irdl.py:441]: Why do you add all three and then only use one?

**Review by webmiche — APPROVED** [review]
LGTM :+1:

File: xdsl/irdl.py (hunk 1 above)
Comment (math-fehr) [reply xdsl/irdl.py:441]: It's for the public API, so we can have these once we need them, and won't need to change this again

**Pull request merged**